eras/shelley/test-suite/bench/Main.hs

```diff
@@ -207,31 +209,38 @@ profileCreateRegPools size = do
 -- ==========================================
 -- Epoch Boundary
 
-profileEpochBoundary :: IO ()
+profileEpochBoundary :: Benchmark
 profileEpochBoundary =
-  defaultMain $
-    [ bgroup "aggregate stake" $
-        epochAt <$> benchParameters
-    ]
+  bgroup "aggregate stake" $ epochAt <$> benchParameters
   where
     benchParameters :: [Int]
-    benchParameters = [10000, 100000, 1000000]
+    benchParameters = [5000, 50000, 500000]
 
 epochAt :: Int -> Benchmark
 epochAt x =
-  env (QC.generate (genTestCase x (10000 :: Int))) $
-    \arg ->
-      bgroup
-        ("UTxO=" ++ show x ++ ",  address=" ++ show (10000 :: Int))
-        [ bench "Using maps" (whnf action2m arg)
-        ]
+  env (QC.generate (genTestCase x n)) $ \arg ->
+    bgroup
+      ("UTxO=" ++ show x ++ ",  address=" ++ show n)
+      [ bench "stakeDistr" (nf action2m arg),
+        bench "incrementalStakeDistr)" (nf action2im arg)
+      ]
+  where
+    n = 10000 :: Int
 
 action2m ::
   ShelleyTest era =>
   (DState (Crypto era), PState (Crypto era), UTxO era) ->
   EB.SnapShot (Crypto era)
 action2m (dstate, pstate, utxo) = stakeDistr utxo dstate pstate
 
+action2im ::
+  ShelleyTest era =>
+  (DState (Crypto era), PState (Crypto era), UTxO era) ->
+  EB.SnapShot (Crypto era)
+action2im (dstate, pstate, utxo) =
+  let incStake = updateStakeDistribution mempty mempty utxo
+   in incrementalStakeDistr incStake dstate pstate
+
```

Comment:
I am unsure what the semantics of "bench" is, is the cost of building  (nf action2im arg) counted? THat might account for the slight increase in  increase in time when using incrementalStakeDistr.  I f that is not the case then I think we have to dig deeper. Something does not look right.